cmd/kubelet/app/server.go

```diff
@@ -263,8 +263,14 @@ HTTP server: The kubelet can also listen for HTTP and respond to a simple API
 			// set up signal context here in order to be reused by kubelet and docker shim
 			ctx := genericapiserver.SetupSignalContext()
 
-			// run the kubelet
+			// make kubelet configuration safe for logging
+			for k := range kubeletServer.KubeletConfiguration.StaticPodURLHeader {
+				kubeletServer.KubeletConfiguration.StaticPodURLHeader[k] = []string{"<redacted>"}
```

Comment:
I'd have to assume this static pod header is important for the kubelet to run below.  You only really want to sanitize this out of the log call, not the entire process, right?

Reply:
Oops. Good point. Should be fixed now.

Reply:
Maps and slices are reference-like.  Passing by value to a function will copy the wrapper, but the underlying references will refer to the same values in memory.  Consequently, this still mutates the `config` passed to this function.

Fortunately, `*KubeletConfiguration` has a `DeepCopy()` method you can use, rather than splitting out a separate method or having to copy things yourself.

Reply:
Once again, oops, and good point. Thank you for catching that.

Switched to DeepCopy.

Reply:
If I were thinking more clearly today, I would have led with this...

A similar concern is addressed in `round_trippers`.  See that `maskValue` approach [here](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/transport/round_trippers.go#L394).  

I don't know how these headers get used, but if you wanted to only redact headers that could possibly contain a bearer token, you could selectively strip the `"Authorization"` header rather than redacting all headers.  Maybe a simple `if _, hasAuth := StaticPodURLHeader["Authorization"] ; hasAuth {` check rather than looping over all keys.

Reply:
Thank you for that suggestion. That sounds like a good idea to me, but I'm not sure how those headers are going to be used since they are entirely user-configurable. In theory, a user could use custom headers, so I think it is safer to mask everything. Maybe this is too heavy handed, though.

In any case, I've changed the hard-coded replacement string to "<masked>" instead of "<redacted>", since "<masked>" is more neutral.

Reply:
(GitHub will try to parse your `<tag>`s, so you have to either `\` escape them or put them in a backtick literal.)

Reply:
Thank you for catching that! For a minute I thought markdown had string literals :upside_down_face: